test/new-e2e/pkg/environments/aws/host/host.go

```diff
@@ -0,0 +1,184 @@
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/fakeintake"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	provisionerBaseID = "aws-ec2vm-"
+	defaultVMName     = "vm"
+)
+
+// ProvisionerParams is a set of parameters for the Provisioner.
+type ProvisionerParams struct {
+	name string
+
+	instanceOptions   []ec2.VMOption
+	agentOptions      []agentparams.Option
+	fakeintakeOptions []fakeintake.Option
+	extraConfigParams runner.ConfigMap
+}
+
+func newProvisionerParams() *ProvisionerParams {
+	// We use nil arrays to decide if we should create or not
+	return &ProvisionerParams{
+		name:              defaultVMName,
+		instanceOptions:   []ec2.VMOption{},
+		agentOptions:      []agentparams.Option{},
+		fakeintakeOptions: []fakeintake.Option{},
+		extraConfigParams: runner.ConfigMap{},
+	}
+}
+
+// ProvisionerOption is a provisioner option.
+type ProvisionerOption func(*ProvisionerParams) error
+
+// WithName sets the name of the provisioner.
+func WithName(name string) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.name = name
+		return nil
+	}
+}
+
+// WithEC2InstanceOptions adds options to the EC2 VM.
+func WithEC2InstanceOptions(opts ...ec2.VMOption) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.instanceOptions = append(params.instanceOptions, opts...)
+		return nil
+	}
+}
+
+// WithAgentOptions adds options to the Agent.
+func WithAgentOptions(opts ...agentparams.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = append(params.agentOptions, opts...)
+		return nil
+	}
+}
+
+// WithFakeIntakeOptions adds options to the FakeIntake.
+func WithFakeIntakeOptions(opts ...fakeintake.Option) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = append(params.fakeintakeOptions, opts...)
+		return nil
+	}
+}
+
+// WithExtraConfigParams adds extra config parameters to the ConfigMap.
+func WithExtraConfigParams(configMap runner.ConfigMap) ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.extraConfigParams = configMap
+		return nil
+	}
+}
+
+// WithoutFakeIntake disables the creation of the FakeIntake.
+func WithoutFakeIntake() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.fakeintakeOptions = nil
+		return nil
+	}
+}
+
+// WithoutAgent disables the creation of the Agent.
+func WithoutAgent() ProvisionerOption {
+	return func(params *ProvisionerParams) error {
+		params.agentOptions = nil
+		return nil
+	}
+}
+
+// ProvisionerNoAgentNoFakeIntake wraps Provisioner with hardcoded WithoutAgent and WithoutFakeIntake options.
+func ProvisionerNoAgentNoFakeIntake(opts ...ProvisionerOption) e2e.TypedProvisioner[environments.Host] {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
Thank you !